docs/adr/0008-reserving-paths-in-publishing-api.md

```diff
@@ -0,0 +1,176 @@
+  *because* it would be a poor user experience if having a title once made it
+  unavailable to use again.
+
+### Could
+
+- allow Content Publisher to reserve base paths without a corresponding document
+  being persisted in the database;\
+  *because* there are unresolved concerns/debates about the point in the
+  workflow that a document is created.
+
+### Won't
+
+- support resolving base path conflicts with a method more
+  sophisticated than a numeric suffix;\
+  *because* there are not decisions/recommendations on a nicer approach and
+  these are therefore considered out of scope.
+- provide means for the slug aspect of a base_path to be any place other than
+  the end of a base_path (e.g. we support `/government/news/:slug` and not
+  `/government/:slug/news`);\
+  *because* only suffix slug base paths are currently used on GOV.UK;
+- consider means that a new document can claim the base path of a document
+  that has a "removed" state on Content Publisher;\
+  *because* claiming old base paths is a niche feature that is complicated
+  due to one document changing the state of another.
+
+## Decision
+
+Two new endpoints should be created in Publishing API:
+
+1. POST `/paths`. This should take attributes of `base_path_prefix`, `title` and
+  `publishing_app`. Given these attributes it will generate and return a base
+   path that does not conflict  with content or existing reservations.
+   Internally the Publishing API will create a `PathReservation`.
+2. DELETE `/path(/*base_path)`. This will remove a `PathReservation` from
+   Publishing API for a given `base_path` if the requesting publishing
+   application is the one that reserved the path.
+
+Content Publisher should call the POST endpoint to reserve paths and it should
+maintain a record of the paths it has reserved. This would be stored on a
+`ReservedPath` model (naming TBC but purposely distinct from `PathReservation`).
+This would store `base_path_prefix`, `title`, `base_path`, and timestamps. This
+model may also store some an id to associate the `ReservedPath` with a
+particular session.
+
+The `Document` model in Content Publisher would store an association to the
+`ReservedPath` that is currently used - it would also be useful to store the one
+the live version of the document is using. The lack of usage by a
+`Document` is an indication that a `ReservedPath` is available for use or can
+be un-reserved from the Publishing API (after a suitable time
+period).
+
+When a user makes changes to the title input on the edit content form a HTTP
+request is sent to Content Publisher to generate a title. This should operate
+as follows:
+
+- The title and path prefix should be used to check whether there is
+  an available `ReservedPath` available in Content Publisher database;
+- If one is found it should be verified that the `ReservedPath` belongs to the
+  corresponding document (or session);
+- If there is not one found the Publishing API should be called to generate
```

Comment:
What happens to the existing `ReservedPath`? do we need to be explicit here that it is kept by Content Publisher?